fastlane/lib/fastlane/actions/get_version_number.rb

```diff
@@ -4,132 +4,80 @@ class GetVersionNumberAction < Action
       require 'shellwords'
 
       def self.run(params)
-        # More information about how to set up your project and how it works:
-        # https://developer.apple.com/library/ios/qa/qa1827/_index.html
-
         folder = params[:xcodeproj] ? File.join(params[:xcodeproj], '..') : '.'
+        scheme = params[:scheme] || ""
+        target_name = params[:target]
+        configuration = params[:configuration]
 
-        command_prefix = [
-          'cd',
-          File.expand_path(folder).shellescape,
-          '&&'
-        ].join(' ')
+        # Get version_number 
+        project = get_project!(folder)
```

Comment:
Aren't `!` used for methods that change the objects? https://stackoverflow.com/a/612196/865175

Reply:
@revolter Also...
>bang is also used for methods that raise an exception when the method without does not

`!` means dangerous. Modifying is dangerous but so is raising an exception 😊

Reply:
Awesome, didn't know 👍